docs/madr/decisions/019-offline-signing-tokens.md

```diff
@@ -0,0 +1,207 @@
+# Offline signing tokens
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/4031
+
+## Context and Problem Statement
+
+Kuma provides multiple tokens for different lines of security
+* User authenticating to Kuma API Server
+* Data plane proxy authenticating to Kuma DP Server
+* Zone proxy authenticating to Kuma DP Server
+
+Right now, a user has to use Kuma CP to issue the tokens.
+While this works, some users may prefer using their existing token issuer.
+We can provide an interface of a token (document required fields) and a way to provide public key of a token.
+This way a user can use different system to issue token, but let Kuma CP validate the token.
+This improves security posture if you already have a safe way of managing signing key for your tokens.
```

Comment:
A slightly related question: Do we currently support rotating dpp token without restarting the dpp? .i.e: if we need to reestablish the xds connection and we rotated the token file will the dpp use the renewed token? If think we did something like this for k8s already no?

Reply:
We do, but it's opt-in, because we had DNS problems with Google gRPC in Envoy
`kuma_dp_server_auth_use_token_path` (apparently missing doc in the config file :/ )